src/webrtc/call.ts

```diff
@@ -0,0 +1,1299 @@
+    for (let i = 0; i < tracks.length; i++) {
+        if (tracks[i].enabled) {
+            return true; // at least one track is enabled
+        }
+    }
+    return false;
+}
+
+function getUserMediaVideoContraints(callType: CallType) {
+    const isWebkit = !!window.navigator.webkitGetUserMedia;
+
+    switch (callType) {
+        case CallType.Voice:
+            return {
+                audio: {
+                    deviceId: audioInput ? {ideal: audioInput} : undefined,
+                }, video: false,
+            };
+        case CallType.Video:
+            return {
+                audio: {
+                    deviceId: audioInput ? {ideal: audioInput} : undefined,
+                }, video: {
+                    deviceId: videoInput ? {ideal: videoInput} : undefined,
+                    /* We want 640x360.  Chrome will give it only if we ask exactly,
+                       FF refuses entirely if we ask exactly, so have to ask for ideal
+                       instead
+                       XXX: Is this still true?
+                     */
+                    width: isWebkit ? { exact: 640 } : { ideal: 640 },
+                    height: isWebkit ? { exact: 360 } : { ideal: 360 },
+                },
+            };
+    }
+}
+
+let audioOutput: string;
+let audioInput: string;
+let videoInput: string;
+/**
+ * Set an audio output device to use for MatrixCalls
+ * @function
+ * @param {string=} deviceId the identifier for the device
+ * undefined treated as unset
+ */
+export function setAudioOutput(deviceId: string) { audioOutput = deviceId; }
+/**
+ * Set an audio input device to use for MatrixCalls
+ * @function
+ * @param {string=} deviceId the identifier for the device
+ * undefined treated as unset
+ */
+export function setAudioInput(deviceId: string) { audioInput = deviceId; }
+/**
+ * Set a video input device to use for MatrixCalls
+ * @function
+ * @param {string=} deviceId the identifier for the device
+ * undefined treated as unset
+ */
+export function setVideoInput(deviceId: string) { videoInput = deviceId; }
+
+/**
+ * Create a new Matrix call for the browser.
+ * @param {MatrixClient} client The client instance to use.
+ * @param {string} roomId The room the call is in.
+ * @param {Object?} options DEPRECATED optional options map.
+ * @param {boolean} options.forceTURN DEPRECATED whether relay through TURN should be
+ * forced. This option is deprecated - use opts.forceTURN when creating the matrix client
+ * since it's only possible to set this option on outbound calls.
+ * @return {MatrixCall} the call or null if the browser doesn't support calling.
+ */
+export function createNewMatrixCall(client: any, roomId: string, options: CallOpts) {
+    // typeof prevents Node from erroring on an undefined reference
+    if (typeof(window) === 'undefined' || typeof(document) === 'undefined') {
+        logger.error("No window or document object: WebRTC is not supported in this environement");
```

Comment:
This change means that when using `matrix-js-sdk` in a non-browser context (e.g. appservice bridges), we get an error every time a client is created, which pollutes the log when nothing really went wrong. Is it possible to lower the log level of this line?